### PR TITLE
[WIP] Remove significant whitespace in hashi_vault tests

### PR DESCRIPTION
**Pull request by markafarrell**
##### SUMMARY
Currently it is required that in hashi_vault tests the `conn_params` variable must end in a space or the tests will fail.

This removes that requirement and adjusts existing tests to remove the space from the end of `conn_params`

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

hashi_vault

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by felixfontein** [comment]
This PR contains #1024, so I guess it will be rebased once #1024 is merged and is this still a WIP? Or should it be standalone PR, to be merged before #1024?

**Comment by markafarrell** [comment]
It's to be rebased and merged after #1024

**Comment by briantist** [comment]
It's been a while since I worked on these tests so apologies if I'm missing something, but to me it kind of seems like removing the space from `conn_params` and adding it to whenever comes next is just as arbitrary as the current case, with the exception that `conn_params` is used very commonly as the first "part" of a concatenated term string. The "significant whitespace" in the that var is just being moved to another place, rather than being removed, if I understand it correctly.

But I'm genuinely interested in your POV on how it's better this way, as I may be overlooking something.

**Comment by markafarrell** [comment]
Maybe it will help a bit if I explain how this tripped me up.

1. Add a new feature to the lookup.
2. Copy an existing test and add my new parameter to the end of connstr
3. Run tests and get an obscure error about `secret` not being set.
4. Searching through lookup code for what broke.
5. Realize that `'secret='` is being concatenated onto the end of connstr
6. Getting annoyed

I think this really just makes it so that people in the future don't get caught like me.

**Comment by briantist** [comment]
Thanks I see what you mean.

I guess the way I'm looking at it is that most changes aren't as likely to change `conn_params`, and that `conn_params` is set once then inherited by the entire include, whereas every new individual call sets something after `conn_params` (usually `secret=` but could be something else), which is a lot more places to potentially miss a space.

I guess it's 6 of one, half a dozen of the other, but the existing way feels a little better to me (I didn't implement that style, so I have no personal attachment).

One of the things I did introduce was the ability to actually set those params as params, and not just concatenated into the term string, specifically to avoid having to do messy concats.

Most of the tests pre-date that PR though, and for backwards compatibility it still has to work the other way, so I tried to keep the old tests and add new ones for the newer style, to keep good coverage.

---

Anyway, in short: I lean slightly toward keeping the space in `conn_params` but your perspective as a recent contributor is important and I don't feel strongly enough to actively oppose changing it or anything like that.

**Comment by felixfontein** [comment]
After merging #1062 there are now conflicts which need to be resolved.

**Comment by dmsimard** [comment]
@briantist FYI -- is this material for merging here or should it be sent to https://github.com/ansible-collections/community.hashi_vault instead ?

**Comment by briantist** [comment]
Hi @markafarrell , continued development of the `hashi_vault` plugin will take place in the [`community.hashi_vault`](https://github.com/ansible-collections/community.hashi_vault) collection, so this PR will be closed in this repo.

As I said I don't feel strongly either way about this change so If you're interested in re-submitting it there, I'll look to include some other recent contributors and go with the community consensus :) thank you for this submission and for #1024 !

close_me
